packages/cli/src/ui/components/DialogManager.tsx

```diff
@@ -46,8 +46,8 @@ export const DialogManager = ({ addItem }: DialogManagerProps) => {
     return (
       <Box borderStyle="round" borderColor={theme.status.warning} paddingX={1}>
         <Text color={theme.status.warning}>
-          Workspace trust has changed. Press &apos;r&apos; to restart Gemini to
-          apply the changes.
+          Workspace trust has changed due to IDE connection or trust change.
+          Press &apos;r&apos; to restart Gemini to apply the changes.
```

Comment:
this message will be confusing to the % of users without an IDE connection. can you make it say why the trust was changed to that it could be IDE connection or trust change if that is trivial enough to plumb through?

Reply:
also please move this out of DialogManager into its own dedicated WorkspaceTrustDialog class for easier testability.

Reply:
Done. This message will never show outside the IDE. It is for IDE connection or IDE trust change. Separated the messages anyway. PTAL.